modules/rostests/rosautotest/CProcess.cpp

```diff
@@ -27,10 +27,11 @@ CProcess::CProcess(const wstring& CommandLine, LPSTARTUPINFOW StartupInfo)
 }
 
 /**
- * Destructs a CProcess object and closes all handles belonging to the process.
+ * Destructs a CProcess object, terminates the process if running, and closes all handles belonging to the process.
  */
 CProcess::~CProcess()
 {
+    TerminateProcess(m_ProcessInfo.hProcess, 255);
```

Comment:
I know it's not nice to ask for extra features, but have you thought about capturing a callstack of the process before calling it?

Reply:
As we both discussed on Mattermost, the current PR can go without that feature and it can be added later.